File: SocialLink.Notifications.Contracts/Details/_NotificationDetails.cs
using SocialLink.SharedKernel.Enumerators;

namespace SocialLink.Notifications.Contracts.Details;

public abstract class NotificationDetails : INotificationDetails
{
	public Guid UserId { get; set; }

	public abstract eNotificationType NotificationTypeId { get; }

	public string Title { get; set; }

	public string Message { get; set; }

	public List<NotificationAction> Actions { get; set; } = [];
}

File: SocialLink.Web.Client/Workspace/src/app/features/notifications/models/details/notification-details.ts
import { eNotificationType } from "../../../../core/enumerators/notification-type.enum";
import { NotificationAction } from "../notification-action";

export class NotificationDetails {
  UserId?: string;
  NotificationTypeId?: eNotificationType;
  Title?: string;
  Message?: string;
  Actions?: NotificationAction[];
}
